metadata-ingestion/src/datahub/ingestion/source/data_lake/__init__.py

```diff
@@ -206,11 +215,19 @@ def read_file(self, file: str) -> Optional[DataFrame]:
             )
         elif file.endswith(".json"):
             df = self.spark.read.json(file)
+        elif file.endswith(".avro"):
+            try:
+                df = self.spark.read.format("avro").load(file)
+            except AnalysisException as e:
+                self.report.report_failure(
+                    file,
+                    "To ingest avro files, please install the spark-avro package: https://mvnrepository.com/artifact/org.apache.spark/spark-avro_2.12/3.0.3",
+                )
+                raise (e)
```

Comment:
We should not raise exception instead `return None`. Raising exception will cause outer loop to exit. But in our sources we would want to skip things and report failures instead of stopping the whole process

Reply:
Ok